src/utils/tokenizers.py

```diff
@@ -53,8 +55,15 @@ def detokenize(self, tokens):
         '''
         return [self._detokenizer.unescape_xml(t) for t in tokens]
 
-
-AVAILABLE_TOKENIZERS = {
-    "OpenAI.BPE": OpenAIBPETokenizer(),
-    "MosesTokenizer": MosesTokenizer()
-}
+@functools.lru_cache(maxsize=8, typed=False)
+def get_tokenizer(tokenizer_name):
+    if tokenizer_name.startswith("bert"):
+        from pytorch_pretrained_bert import BertTokenizer
+        do_lower_case = tokenizer_name.endswith('uncased')
+        tokenizer = BertTokenizer.from_pretrained(tokenizer_name, do_lower_case=do_lower_case)
+    else:
+        # return the class instantiation given class name
+        assert tokenizer_name in globals(), "Tokenizer name not found in tokenizers available!"
```

Comment:
Using `globals()` is **strongly** discouraged (this will lead to all kinds of bugs you don't expect). 

Just put some if/else statements here? We're not adding new tokenizers that often so there's not a big need to avoid a couple lines of boilerplate. Plus, this way it's easy to do lazy package imports like we do for BERT.

Reply:
Gotcha - I was going to do simple if-else here but thought to reduce boilerplate.